app/com/gu/floodgate/contentsource/ContentSourceApi.scala

```diff
@@ -66,12 +70,27 @@ class ContentSourceApi(contentSourceService: ContentSourceService, reindexServic
     }
   }
 
+  def cancelReindex(id: String) = Action.async { implicit request =>
+    reindexService.cancelReindex(id) map { happyOrError =>
+      happyOrError match {
+        case Good(_) => Ok
+        case Bad(error) => BadRequest(Json.toJson(ErrorResponse(error.message)))
```

Comment:
Splitting hairs, but I'm not sure if `BadRequest` is the best response code here. It's hard to say whether the failure is the client's or the server's fault, so maybe `InternalServerError` would be more appropriate.